src/Nethermind/Nethermind.Synchronization/FastBlocks/SyncStatusList.cs

```diff
@@ -30,43 +30,43 @@ public void GetInfosForBatch(BlockInfo?[] blockInfos)
             int collected = 0;
 
             long currentNumber = LowestInsertWithoutGaps;
-            lock (_statuses)
+            while (collected < blockInfos.Length && currentNumber != 0)
```

Comment:
ok this can be called highly multithreaded by `PrepareRequest`, it would be better that it return a continuous range of blocks, now it is easy to make it non-continuous, not sure if worth it.